lib/semantics/symbol.h

```diff
@@ -304,14 +304,14 @@ class NamelistDetails {
 
 class CommonBlockDetails {
 public:
-  std::list<Symbol *> &objects() { return objects_; }
-  const std::list<Symbol *> &objects() const { return objects_; }
-  void add_object(Symbol &object) { objects_.push_back(&object); }
+  std::list<SymbolRef> &objects() { return objects_; }
+  const std::list<SymbolRef> &objects() const { return objects_; }
+  void add_object(const Symbol &object) { objects_.emplace_back(object); }
   MaybeExpr bindName() const { return bindName_; }
   void set_bindName(MaybeExpr &&expr) { bindName_ = std::move(expr); }
 
 private:
-  std::list<Symbol *> objects_;
+  std::list<SymbolRef> objects_;
```

Comment:
I'm not sure there is a good reason this is a list rather than a vector, so we could use `SymbolVector` here.

Reply:
`SymbolVector` works (I tried it).